src/julia_gc.c

```diff
@@ -605,7 +634,7 @@ static void GapRootScanner(int full)
     for (Int i = 0; i < GlobalCount; i++) {
         Bag p = *GlobalAddr[i];
         if (IS_BAG_REF(p)) {
-            JMark(p);
+            JMarkGapObjSafe(p);
```

Comment:
Why does this need to use `JMarkGapObjSafe`? It seems to be me that we even could use `jl_gc_mark_queue_obj` directly here, no?

Reply:
It probably could. I'll have a look at that.